### PR TITLE
Make exportdb tool faster

### LINKED ISSUE #9141: Make exportdb tool faster
# 🚀 Feature Request

### Description

exportdb tool takes long time to export the stats if the db is large. This issue is t make it faster.

### Describe the solution you'd like

- Parallelize the bucket processing (process every bucket in its own routine)
- If there are more keys, process them parallely as several block.

### Describe alternatives you've considered

 Have you considered any alternative solutions or workarounds?

### PR DESCRIPTION
**Pull request by jmozah**
**What type of PR is this?**
Performance Enhancement

**What does this PR do? Why is it needed?**
This PR is supposed to make the processing and printing of DB information faster.
Especially if the DB is a bigger one.
- [x] Parallelise the bucket processing (process every bucket in its own routine)
- [x] If there are more keys, process them parallely as several block.

**Which issues(s) does this PR fix?**
Fixes #9141

**Other notes for review**
How to run the stats
- go run main.go --datadir `/data1/prysm-data-mainnet/beaconchaindata` --dbname `beaconchain.db` --bucket-stats=`true`
- go run main.go --datadir `/data1/prysm-data-mainnet/beaconchaindata` --dbname `beaconchain.db` --bucket-contents=`true` --bucket-name=`state` --limit=`1000`

### CODE REVIEW

File: tools/exploredb/main.go [hunk 1]
```diff
@@ -95,143 +175,166 @@ func showBucketStats(dbNameWithPath string) {
...
-	// dont forget to close it when ejecting out of this function.
-	defer func() {
-		closeErr := db.Close()
-		if closeErr != nil {
-			log.Fatalf("could not close db, %v", closeErr)
+func readStateSummary(ctx context.Context, db *kv.Store, stateSummaryC chan<- *modifiedStateSummary, keys [][]byte, sizes []uint64) {
+	for rowCount, key := range keys {
+		ss, ssErr := db.StateSummary(ctx, bytesutil.ToBytes32(key))
+		if ssErr != nil {
+			log.Errorf("could not get state summary for key : , %v", ssErr)
```
Comment (prestonvanloon) [inline tools/exploredb/main.go:260]: A common pattern we use is the `WithError(error)` method with logrus. It adds the error as field metadata and makes it easier to query these logs in a deployed environment. 
```suggestion
			log.WithError(ssErr).Error("could not get state summary for key")
```

File: tools/exploredb/main.go [hunk 2]
```diff
@@ -95,143 +175,166 @@ func showBucketStats(dbNameWithPath string) {
...
+func printBucketStates(statsC <-chan *bucketStat, doneC chan<- bool) {
+	for stat := range statsC {
+		if stat.noOfRows != 0 {
+			averageValueSize := stat.totalValueSize / stat.noOfRows
+			averageKeySize := stat.totalKeySize / stat.noOfRows
+			fmt.Println("------ ", stat.bucketName, " --------")
+			fmt.Println("NumberOfRows     = ", stat.noOfRows)
+			fmt.Println("TotalBucketSize  = ", humanize.Bytes(stat.totalValueSize+stat.totalKeySize))
+			fmt.Println("KeySize          = ", humanize.Bytes(stat.totalKeySize), "(min = "+humanize.Bytes(stat.minKeySize)+", avg = "+humanize.Bytes(averageKeySize)+", max = "+humanize.Bytes(stat.maxKeySize)+")")
+			fmt.Println("ValueSize        = ", humanize.Bytes(stat.totalValueSize), "(min = "+humanize.Bytes(stat.minValueSize)+", avg = "+humanize.Bytes(averageValueSize)+", max = "+humanize.Bytes(stat.maxValueSize)+")")
```
Comment (prestonvanloon) [inline tools/exploredb/main.go:283]: Can we use `log` for these?

Comment (prestonvanloon) [reply tools/exploredb/main.go:283]: Same comment for the rest of the tool. Logrus gives a bit more flexibility with the log output formats.

File: tools/exploredb/main.go (hunk 1 above)
Comment (jmozah) [reply tools/exploredb/main.go:260]: cool. Will use this in mind for future too.

File: tools/exploredb/main.go (hunk 2 above)
Comment (jmozah) [reply tools/exploredb/main.go:283]: Sure.

**Review by rauljordan — DISMISSED** [review]

**Review by rauljordan — APPROVED** [review]

**Pull request merged**